tds_test.go

```diff
@@ -662,6 +662,50 @@ func TestSecureConnection(t *testing.T) {
 	}
 }
 
+func TestTDS8Connection(t *testing.T) {
+	checkConnStr(t)
+	tl := testLogger{t: t}
+	defer tl.StopLogging()
+	SetLogger(&tl)
+
+	dsn := makeConnStr(t)
+	if !strings.HasSuffix(strings.Split(dsn.Host, ":")[0], ".database.windows.net") {
+		t.Skip()
+	}
+	dsnParams := dsn.Query()
+	dsnParams.Set("encrypt", "strict")
+	dsnParams.Set("TrustServerCertificate", "false")
+	dsnParams.Set("tlsmin", "1.2")
+	dsn.RawQuery = dsnParams.Encode()
+
+	conn, err := sql.Open("mssql", dsn.String())
```

Comment:
>sql.Open("mssql", d [](http://example.com/codeflow?start=14&length=19)

this won't handle connecting with AAD auth, which is what the ADO tests use.

Reply:
to require an Azure connection, see how `azuread/azuread_test.go` does it.

Reply:
Moved the test to azuread_test,go and modified this one to ensure `TrustServerCertificate` does not work when `encrypt` is set to `strict`